lib/interfaces.ts

```diff
@@ -36,15 +46,15 @@ export interface IModuleOptions<
   stateInit?: () => S
   getters?: (state: S) => G
   mutations?: (state: S) => M
-  plugins?: IPlugin<S>[]
+  plugins?: (ICreatePlugin<S> | IPlugin<S>)[]
```

Comment:
Updated plugins to allow initializing with the module as a parameter. It's useful for allowing plugins to change their behavior based on the module name/version/etc.